### PR TITLE
Fix bug 1443551: List custom plural forms

### PR DESCRIPTION
**Pull request by mathjazz**
If source string contains custom (numeric) plural forms, list them as variants in untranslated, along with locale CLDR plurals.

Test case:
https://github.com/mozilla-l10n/pontoon-ftl/blob/master/en-US/demo.ftl#L30

### CODE REVIEW

File: pontoon/base/static/js/fluent_interface.js [hunk 1]
```diff
@@ -77,6 +77,20 @@ var Pontoon = (function (my) {
...
+   * The list if a union of:
+   *   - any custom (numeric) plural forms specified in the original string (bug 1443551)
+   *   - CLDR plurals of the locale
+   */
+  function getPluralForms(element) {
+    var customPluralForms = element.expression.variants.map(function (variant) {
+      return variant.key.value;  // can be undefined, so we filter it out below
+    });
+
+    return customPluralForms.filter(Boolean).concat(Pontoon.locale.cldr_plurals);
```
Comment (Pike) [inline pontoon/base/static/js/fluent_interface.js]: This filters out `0`, too?

Also, would this add `one` to locales that only have `[many]` as plural forms?

Comment (Pike) [inline pontoon/base/static/js/fluent_interface.js]: You probably want to exclude the en-US plurals, and you don't want to filter out `0`.

Comment (mathjazz) [reply pontoon/base/static/js/fluent_interface.js]: Thanks! It doesn't include en-US plurals (that would be `item.key.name`), but it definitely filters out 0. I'll fix that and make the code more explicit.

File: pontoon/base/static/js/fluent_interface.js [hunk 2]
```diff
@@ -77,6 +77,25 @@ var Pontoon = (function (my) {
...
+   * The list if a union of:
+   *   - any custom (numeric) plural forms specified in the original string (bug 1443551)
+   *   - CLDR plurals of the locale
+   */
+  function getPluralForms(element) {
+    var customPluralForms = element.expression.variants.filter(function(item) {
+      if (item.key.type === 'NumberLiteral') {
+        return true;
+      }
+      return false;
```
Comment (Pike) [inline pontoon/base/static/js/fluent_interface.js]: Just `return item.key.type === 'NumberLiteral';` ?

**Pull request merged**